crates/context_aware_config/src/api/dimension/handlers.rs

```diff
@@ -101,9 +122,31 @@ async fn create(
 }
 
 #[get("")]
-async fn get(db_conn: DbConnection) -> superposition::Result<Json<Vec<Dimension>>> {
+async fn get(
+    db_conn: DbConnection,
+    state: Data<AppState>,
+    tenant: Tenant,
+) -> superposition::Result<HttpResponse> {
     let DbConnection(mut conn) = db_conn;
 
     let result: Vec<Dimension> = dimensions.get_results(&mut conn)?;
-    Ok(Json(result))
+
+    let dimensions_vec: Vec<Map<String, Value>> = result
+        .into_iter()
+        .filter_map(|ele| {
+            serde_json::to_value(&ele)
+                .ok()
+                .and_then(|value| value.as_object().cloned())
+        })
+        .collect();
+
+    let mandatory_dimensions =
+        get_mandatory_dimensions(&state.mandatory_dimensions, tenant);
+
+    let new_map: Vec<Map<String, Value>> = dimensions_vec
+        .into_iter()
+        .map(|ele| insert_mandatory_dimension(ele, mandatory_dimensions.clone()))
```

Comment:
instead of doing all these conversions (in the newly code) to insert the field `mandatory`
it be better if we create a separate type for Dimension which includes the field mandatory as well, that way the existence of the field `mandatory` gets documented as well along with being able to make this code cleaner by implementing a From trait for the newly added type from the already existing Dimension type

something like:
```suggestion
        let mandatory_dimensions =
               get_mandatory_dimensions(&state.mandatory_dimensions, tenant);
        result.into_iter().map(DimensionWithMandatory::from).map(|d| {
              DimensionWithMandatory { mandatory: mandatory_dimensions.contains(d.dimension), ..d}
        })
        .collect();
```

the from trait can be something like:
```
impl From<Dimension> for DimensionWithMandatory {
    fn from(value: Dimension) -> Self {
        DimensionWithMandatory {
            mandatory: false,
            ..value.into()
        }
    }
}
```

this setup seems cleaner than what we have right now, with all the conversions and getter around Value type